### ISSUE TITLE
Location of warnings in quickfix window for multi-file document always set to main file

### ISSUE THREAD

**Issue by gravndal** [lervag/vimtex#1919]
**Issue**
When compiling a multi-file document with a warning, vimtex always thinks it is in the main tex-file.

**main.tex**
```tex
\documentclass{article}
\begin{document}
\input{import.tex}
\end{document}
```

**import.tex**
```tex
\begin{verbatim}
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
\end{verbatim}
```

**Commands/Input**
1. `$ XDG_CONFIG_HOME=$HOME/vimtex-repro/config/ XDG_DATA_HOME=$HOME/vimtex-repro/data/ nvim -u NORC main.tex`
2. `<localleader>ll`

Note: config is empty, data contains a clone of vimtex under `site/pack/_/start`

**Observed Behaviour**
```
main.tex|3 warning| Overfull \hbox (179.99542pt too wide) in paragraph at lines 3--3
```

Relevant part of compile output:
```
 (./import.tex
Overfull \hbox (179.99542pt too wide) in paragraph at lines 3--3
[]\OT1/cmtt/m/n/10 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa[] 
)
```

**Expected Behaviour**
Correctly parsed context for warnings (much like errors).

**Output from VimtexInfo**
```
System info
  OS: Linux 5.10.5
  Vim version: NVIM v0.5.0-dev
  Has clientserver: true
  Servername: /tmp/nvimqBYAHi/0

vimtex project: main
  base: main.tex
  root: /home/r/vimtex-repro
  tex: /home/r/vimtex-repro/main.tex
  out: /home/r/vimtex-repro/main.pdf
  log: /home/r/vimtex-repro/main.log
  aux: /home/r/vimtex-repro/main.aux
  fls: /home/r/vimtex-repro/main.fls
  main parser: current file verified
  source files:
    main.tex
    import.tex
  compiler: latexmk
    configuration: 
      continuous: 1
      callback: 1
      latexmk options:
        -verbose
        -file-line-error
        -synctex=1
        -interaction=nonstopmode
      latexmk engine: -pdf
    job: 3
    pid: 1507433
    cmd: max_print_line=2000 latexmk -verbose -file-line-error -synctex=1 -interaction=nonstopmode -pdf -pvc -e '$success_cmd = "echo vimtex_compiler_callback_success"' -e '$failure_cmd = "echo vimtex_compiler_callback_failure"' 'main.tex'
  viewer: General
  qf: LaTeX logfile
    root: /home/r/vimtex-repro
    addqflist: 57
    errorformat_saved: 
    fix_paths: 58
    main: /home/r/vimtex-repro/main.tex
    set_errorformat: 56
  document class: article
```

**Comment by lervag** [comment]
I don't think this is quite possible to fix. Or, at least, it is highly untrivial. The problem is that `\input` actually includes the content during compilation so the error message itself points to the `main.tex` file (see for yourself by inspecting the `.log` file). If you use `\include`, it should work.

I think one approach (that I have not considered before) would be to map the entire tex project manually. I.e., parse all tex files to create one continuous "file" entity without the `\input` and `\include` lines. This parsed content could be used to find the correct file and line, possibly. But it does seem difficult and it also seems like a lot of work for not that much gain. So I'm not sure I want to go that path...

**Comment by gravndal** [comment]
I'm seeing the same issue when using `\include`:
![2021-01-12-08:51:39](https://user-images.githubusercontent.com/4656860/104285518-0e8e8600-54b4-11eb-8e68-33069fb010fb.png)
Though in the log there is that extra ```\openout2 = `import.aux'.``` which I'm guessing you're referring to?

**Comment by lervag** [comment]
Did you remove the old files first?

**Comment by gravndal** [comment]
Yes.

**Comment by lervag** [comment]
Ah, I can reproduce; sorry. If you changed the `aaa`'s to `\xxx`, you will see it now works. The "Overfull \hbox" warning is not so easy to parse, unfortunately. See e.g. [this tex.stackexchange.com question](https://tex.stackexchange.com/questions/235051/an-equivalent-of-file-line-error-for-warnings#235064).

So, you might ask, why not just parse the `(./import.tex` line above. The reason is that there are a lot of files that are parsed like this, often with very long file names. For some archaic and silly reason, long filenames are broken in the middle when the length exceeds some specified width (perhaps 80). I've found no way to stop this (even the `max_print_line=2000` does not fully work as expected). Further, I've found no simple and reliable way to reconstruct the file names. To parse the file names we have to be able to parse all of them, else things break unexpectedly. And there it is; I've not found a good solution to this. Not saying it is impossible.

What you could try is to add an external parser with `g:vimtex_quickfix_method` (see the docs). With `pplatex`, this particular case works well for me. But this might work less well for other types of warnings/errors.

**Comment by gravndal** [comment]
`\anerror` actually gets parsed to the correct file with both `\input` and `\include`, but I'll give pplatex a try either way.

**Comment by lervag** [comment]
Ah, sorry, it seems my initial comment was wrong then. Sorry about that. I hope you find `pplatex` to work well.
